### DISCUSSION TITLE
Route discovery

### DISCUSSION THREAD

**Discussion by synologic** [apache/apisix#8596] [General]
Hi all,

I have some difficulties trying to figure out from the documentation if a scenario is achievable, namely dynamic route discovery.

I am looking for a way to be able to have discoverable routes based on registration from the backend servers (consul, eureka, doesn't matter)

Basically, when a backend server (service) starts and attempts to "register" a route say GET /service1 this would be automatically discoveared.

Any way to have this ?

**Comment by juzhiyuan** [comment]
Hi, just to confirm: do you mean how to use APISIX with Service Discovery components? 🤔 Which doc you're reading?

**Comment by synologic** [comment]
Not service discovery, route discovery, for example, I start a service which is discoverable by whatever (say Consul) but that service publishes a new route, a unique route that should be configured automatically and not manually
